.github/workflows/dev_nifi.yaml

```
@@ -0,0 +1,113 @@
+---
+name: Build and publish NiFi for signal-processing demo
+
+env:
+  IMAGE_NAME: nifi
+  IMAGE_VERSION: 1.27.0-postgresql
+  REGISTRY_PATH: demos
+  DOCKERFILE_PATH: "../demos/signal-processing/Dockerfile-nifi"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - demos/signal-processing/Dockerfile-nifi
+      - .github/workflows/dev_nifi.yaml
+
+jobs:
+  build:
+    name: Build/Publish ${{ matrix.runner.arch }} Image
+    permissions:
+      id-token: write
+    runs-on: ${{ matrix.runner.name }}
+    strategy:
+      matrix:
+        runner:
+          - {name: "ubuntu-latest", arch: "amd64"}
+          - {name: "ubicloud-standard-8-arm", arch: "arm64"}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be # v1.3.1
+        with:
+          # This might remove tools that are actually needed, if set to "true" but
+          # frees about 6 GB.
+          tool-cache: false
+
+          # All of these default to true, but feel free to set to "false" if
+          # necessary for your workflow.
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: true
+          swap-storage: true
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@988b5a0280414f521da01fcc63a27aeeb4b104db # v3.6.1
+
+      - name: Building ${{ env.IMAGE_NAME }}
+        shell: bash
+        run: echo ${{ env.IMAGE_NAME }}
+
+      - name: Build image
+        env:
+          IMAGE_REPOSITORY: ${{ env.IMAGE_NAME }}
+          IMAGE_ARCH: ${{ matrix.runner.arch }}
+        shell: bash
+        run: |
+          set -euo pipefail
+          docker buildx build -f "${{ env.DOCKERFILE_PATH }}" --platform "linux/${{ env.IMAGE_ARCH }}" --tag "registry-placeholder/${{ env.REGISTRY_PATH }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_VERSION }}" .
+          echo "IMAGE_ARCH=${{ env.IMAGE_ARCH }}" >> "$GITHUB_ENV"
```

suggestion: Let's combine these two steps.

```suggestion
      - name: Build Demo Image
        env:
          IMAGE_REPOSITORY: ${{ env.IMAGE_NAME }}
          IMAGE_ARCH: ${{ matrix.runner.arch }}
        shell: bash
        run: |
          set -euo pipefail
          echo ${{ env.IMAGE_NAME }}
          docker buildx build -f "${{ env.DOCKERFILE_PATH }}" --platform "linux/${{ env.IMAGE_ARCH }}" --tag "registry-placeholder/${{ env.REGISTRY_PATH }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_VERSION }}" .
          echo "IMAGE_ARCH=${{ env.IMAGE_ARCH }}" >> "$GITHUB_ENV"
```

---

https://github.com/stackabletech/demos/pull/103/commits/a995b9f306a922af35fecbfffb5790861b60bd3b